internal/ir/xds.go

```diff
@@ -445,6 +445,26 @@ type TLSCACertificate struct {
 	Certificate []byte `json:"certificate,omitempty" yaml:"certificate,omitempty"`
 }
 
+// SubjectAltName holds the subject alternative name for the certificate
+// This is the internal representation of the SubjectAltName in the Gateway API
+// https://github.com/kubernetes-sigs/gateway-api/blob/6fbbd90954c2a30a6502cbb22ec6e7f3359ed3a0/apis/v1alpha3/backendtlspolicy_types.go#L177
+// +k8s:deepcopy-gen=true
+type SubjectAltName struct {
+	// Type determines the format of the Subject Alternative Name. Always required.
+	// It is one of:
+	// - "DNS"
+	// - "URI"
+	Type string `json:"type" yaml:"type"`
```

Comment:
* can we omit `Type` to save space, and convert `Hostname` and `URI` to pointers 
 * the switch logic can be moved into the gateway api layer, so if we add any additional validation failure, it can be reported back in the status

Reply:
Done